dataconfig/parsers.py

```diff
@@ -68,7 +69,32 @@ def from_yaml(cls, yaml_path: Union[str, Path]) -> _ConfigIO:
         # FIXME: type checking is ignored for the return statement because mypy
         # doesn't seem to know this is an abstract base class, and the argument
         # unpacking makes sense when instantiating any of the derived classes.
-        return cls(**read_yaml(yaml_path))  # type: ignore
+
+        # read in the config
+        conf = read_yaml(yaml_path)
+        
+        # get the leafs of the config file -> these are the values we potentially need to apply the transformations to
+        # leaves is an Iterable[_Path_t]
+        leaves = _leaves(_nodes(conf))
+
+        # create paths for access of class member leaves
+        cls_leave_trans = map(
+            lambda p: ('.__annotations__.'.join(('',) + p) + '.transformation')[1:],
+            leaves
+        )
```

Comment:
This may not be needed with the above changes

Reply:
Okay, I think you are partly right. The keys are indeed not directly available, not sure if `__annotations__` is the correct place to access it, but for now let's go with that.  The problem is the transformation is normally accessible so you traverse through `__annotations__`, but then you check directly with the class, `cls.mytransform` for the transformation.  Does that makes sense?  Kinda like `cls.__annotations__[key1].__annotations__[key2].mytransform`; as a glom spec that would be: `__annotations__.key1.__annotations__.key2.mytransform`

Maybe go with something like:

    token = "__annotations__." # note the trailing dot
    glom_spec = f"{token}{token.join(path).mytransform"